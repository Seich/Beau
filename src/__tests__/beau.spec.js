const yaml = require('js-yaml')
const Beau = require('../beau')
const { moduleVersion } = require('../shared')

jest.mock('../shared')

const requireg = require('requireg')
requireg.resolving = false

describe(`Beau's config Loader.`, () => {
    it('should load the config', () => {
        moduleVersion.mockReturnValue(1)

        const doc = yaml.load(`
            version: 1
            endpoint: 'http://example.com'

            defaults:
                headers:
                    authentication: hello
        `)

        const beau = new Beau(doc)
        expect(beau).toMatchSnapshot()
    })

    it(`should load the request list using the configuration`, () => {
        moduleVersion.mockReturnValue(1)

        const doc = yaml.load(`
            version: 1
            endpoint: 'http://example.com'

            GET /posts/1: get-post
            GET /user:
                alias: user
                headers:
                    hello: world
        `)

        const beau = new Beau(doc)
        expect(beau.requests).toMatchSnapshot()
    })

    it('should display a warning if the module version and the beau file version are different', () => {
        let stdout
        let spy = jest
            .spyOn(console, 'warn')
            .mockImplementation((val) => (stdout = val))

        moduleVersion.mockReturnValue(2)

        const beau = new Beau({ version: 1 })
        expect(stdout).toEqual('This Beau file expects v1. You are using v2.')

        spy.mockReset()
        spy.mockRestore()
    })
})
