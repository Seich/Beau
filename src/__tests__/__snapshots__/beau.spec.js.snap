// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Beau's config Loader. should create a request list 1`] = `
Beau {
  "config": Config {
    "CACHE": false,
    "DEFAULTS": Object {
      "headers": Object {
        "authentication": "hello",
      },
    },
    "ENDPOINT": "http://jsonplaceholder.typicode.com",
    "ENVIRONMENT": Object {},
    "HOSTS": Array [],
    "PLUGINS": Plugins {
      "context": Object {},
      "registry": Object {
        "dynamicValues": Array [],
        "postRequestModifiers": Array [],
        "preRequestModifiers": Array [],
      },
    },
    "REQUESTS": Array [
      Object {
        "ALIAS": "get-post",
        "ENDPOINT": "http://jsonplaceholder.typicode.com",
        "HEADERS": Object {
          "authentication": "hello",
        },
        "REQUEST": "GET /posts/1",
      },
      Object {
        "ALIAS": "user",
        "ENDPOINT": "http://jsonplaceholder.typicode.com",
        "HEADERS": Object {
          "authentication": "hello",
          "hello": "world",
        },
        "REQUEST": "GET /user",
      },
    ],
    "VERSION": 1,
    "configKeys": Array [
      "VERSION",
      "CACHE",
      "ENDPOINT",
      "PLUGINS",
      "DEFAULTS",
      "ENVIRONMENT",
      "HOSTS",
    ],
    "defaultConfigValues": Object {
      "CACHE": false,
      "DEFAULTS": Object {},
      "ENDPOINT": "",
      "ENVIRONMENT": Object {},
      "HOSTS": Array [],
      "PLUGINS": Array [],
      "VERSION": 1,
    },
    "doc": Object {
      "GET /posts/1": "get-post",
      "GET /user": Object {
        "alias": "user",
        "headers": Object {
          "hello": "world",
        },
      },
      "defaults": Object {
        "headers": Object {
          "authentication": "hello",
        },
      },
      "endpoint": "http://jsonplaceholder.typicode.com",
      "version": 1,
    },
  },
  "requests": RequestList {
    "REQUESTS": Array [
      Object {
        "ALIAS": "get-post",
        "ENDPOINT": "http://jsonplaceholder.typicode.com",
        "HEADERS": Object {
          "authentication": "hello",
        },
        "REQUEST": "GET /posts/1",
      },
      Object {
        "ALIAS": "user",
        "ENDPOINT": "http://jsonplaceholder.typicode.com",
        "HEADERS": Object {
          "authentication": "hello",
          "hello": "world",
        },
        "REQUEST": "GET /user",
      },
    ],
    "cache": RequestCache {
      "$cache": Object {
        "env": Object {},
      },
    },
    "config": Config {
      "CACHE": false,
      "DEFAULTS": Object {
        "headers": Object {
          "authentication": "hello",
        },
      },
      "ENDPOINT": "http://jsonplaceholder.typicode.com",
      "ENVIRONMENT": Object {},
      "HOSTS": Array [],
      "PLUGINS": Plugins {
        "context": Object {},
        "registry": Object {
          "dynamicValues": Array [],
          "postRequestModifiers": Array [],
          "preRequestModifiers": Array [],
        },
      },
      "REQUESTS": Array [
        Object {
          "ALIAS": "get-post",
          "ENDPOINT": "http://jsonplaceholder.typicode.com",
          "HEADERS": Object {
            "authentication": "hello",
          },
          "REQUEST": "GET /posts/1",
        },
        Object {
          "ALIAS": "user",
          "ENDPOINT": "http://jsonplaceholder.typicode.com",
          "HEADERS": Object {
            "authentication": "hello",
            "hello": "world",
          },
          "REQUEST": "GET /user",
        },
      ],
      "VERSION": 1,
      "configKeys": Array [
        "VERSION",
        "CACHE",
        "ENDPOINT",
        "PLUGINS",
        "DEFAULTS",
        "ENVIRONMENT",
        "HOSTS",
      ],
      "defaultConfigValues": Object {
        "CACHE": false,
        "DEFAULTS": Object {},
        "ENDPOINT": "",
        "ENVIRONMENT": Object {},
        "HOSTS": Array [],
        "PLUGINS": Array [],
        "VERSION": 1,
      },
      "doc": Object {
        "GET /posts/1": "get-post",
        "GET /user": Object {
          "alias": "user",
          "headers": Object {
            "hello": "world",
          },
        },
        "defaults": Object {
          "headers": Object {
            "authentication": "hello",
          },
        },
        "endpoint": "http://jsonplaceholder.typicode.com",
        "version": 1,
      },
    },
    "list": Array [
      Request {
        "ALIAS": "get-post",
        "DEPENDENCIES": Set {},
        "ENDPOINT": "http://jsonplaceholder.typicode.com",
        "FORM": undefined,
        "HEADERS": Object {
          "authentication": "hello",
        },
        "PARAMS": undefined,
        "PATH": "/posts/1",
        "PAYLOAD": undefined,
        "VERB": "GET",
        "originalRequest": Object {
          "ALIAS": "get-post",
          "ENDPOINT": "http://jsonplaceholder.typicode.com",
          "HEADERS": Object {
            "authentication": "hello",
          },
          "REQUEST": "GET /posts/1",
        },
        "plugins": Plugins {
          "context": Object {},
          "registry": Object {
            "dynamicValues": Array [],
            "postRequestModifiers": Array [],
            "preRequestModifiers": Array [],
          },
        },
      },
      Request {
        "ALIAS": "user",
        "DEPENDENCIES": Set {},
        "ENDPOINT": "http://jsonplaceholder.typicode.com",
        "FORM": undefined,
        "HEADERS": Object {
          "authentication": "hello",
          "hello": "world",
        },
        "PARAMS": undefined,
        "PATH": "/user",
        "PAYLOAD": undefined,
        "VERB": "GET",
        "originalRequest": Object {
          "ALIAS": "user",
          "ENDPOINT": "http://jsonplaceholder.typicode.com",
          "HEADERS": Object {
            "authentication": "hello",
            "hello": "world",
          },
          "REQUEST": "GET /user",
        },
        "plugins": Plugins {
          "context": Object {},
          "registry": Object {
            "dynamicValues": Array [],
            "postRequestModifiers": Array [],
            "preRequestModifiers": Array [],
          },
        },
      },
    ],
  },
}
`;
